crates/turbopack-ecmascript/src/typescript/mod.rs

```diff
@@ -115,7 +115,31 @@ impl Asset for TsConfigModuleAsset {
                 }
             })
             .await?;
-            let types = types.unwrap_or_else(|| vec![(self.source, "node".to_string())]);
+            let types = if let Some(types) = types {
+                types
+            } else {
+                let mut all_types = Vec::new();
+                let mut current = self.source.path().parent().resolve().await?;
+                loop {
+                    if let DirectoryContent::Entries(entries) =
+                        &*current.join("node_modules/@types").read_dir().await?
```

Comment:
This should be its own function, like `traverse_node_modules`, which would return an async iterator, but that's not needed for now.